portal-user-models/src/main/java/com/telecominfraproject/wlan/portaluser/models/PortalUser.java

```diff
@@ -79,18 +86,35 @@ public void setPassword(String password) {
 		this.password = password;
 	}
 
-	public PortalUserRole getRole() {
-		return role;
+	public List<PortalUserRole> getRoles() {
+		return roles;
 	}
 
-	public void setRole(PortalUserRole role) {
-		this.role = role;
+	public void setRoles(List<PortalUserRole> roles) {
+		// This is to cover the backwards-compatibility case when the
+		// roles property is not provided, but the role property is provided.
+		// This check can be removed when the deprecated getRole/setRole are
+		// removed.
+		if (roles != null) {
+			this.roles = roles;
+		}
 	}
 
-	public void setDetails(PortalUserDetails details) {
-		this.details = details;
+	@JsonIgnore
```

Comment:
this @JsonIgnore will need to be removed - we added special code to deal with the situation when both role and roles properties are provided